README.md

```diff
@@ -102,23 +105,104 @@ Environment:
-`channels/<control_channel>/messages/services/<service_name>/<subtopic>`  
+* `channels/<control_channel_id>/messages/services/<service_name>/<subtopic>`  
+
 when messages is received Agent forwards them to Nats on subject:   
-`commands.<service_name>.<subtopic>`.  
+* `commands.<service_name>.<subtopic>`.  
+
 Payload is up to the application and service itself.
 
 Example of on command can be:
 
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/services/adc -h <mqtt_host> -p 1883  -m  "[{\"bn\":\"1:\", \"n\":\"read\", \"vs\":\"temperature\"}]"
 ```
-mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel>/messages/services/adc -h <mqtt_host> -p 1883  -m  "[{\"bn\":\"1:\", \"n\":\"read\", \"vs\":\"temperature\"}]"
+
+## Heartbeat service
+Services running on the same host can publish to `heartbeat.<service-name>.<service-type>` a heartbeat message.  
+Agent will keep a record on those service and update their `live` status.
+If heartbeat is not received in 10 sec it marks it `offline`.
+Upon next heartbeat service will be marked `online` again.
+
+To check services that are currently registered to agent you can:
+
+```bash
+curl -s -S X GET http://localhost:9000/services
+```
+```json
+[
+  {
+    "name": "duster",
+    "last_seen": "2020-04-28T18:06:56.158130519+02:00",
+    "status": "offline",
+    "type": "test",
+    "terminal": 0
+  },
+  {
+    "name": "scrape",
+    "last_seen": "2020-04-28T18:06:39.58849766+02:00",
+    "status": "offline",
+    "type": "test",
+    "terminal": 0
+  }
+]
 ```
 
+Or you can send a command via MQTT to Agent and receive response on MQTT topic like this:
+
+In one terminal subscribe for result:
+
+```bash
+mosquitto_sub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h <mqtt_host> -p 1883  
+```
+
+In another terminal publish request to view the list of services:
+
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h <mqtt_host> -p 1883  -m  '[{"bn":"1:", "n":"config", "vs":"view"}]'  
+```
+
+Check the output in terminal where you subscribed for results. You should see something like:
+
+```json
+[
+  {
+    "bn": "1",
+    "n": "view",
+    "t": 1588091188.8872917,
+    "vs": "[{\"name\":\"duster\",\"last_seen\":\"2020-04-28T18:06:56.158130519+02:00\",\"status\":\"offline\",\"type\":\"test\",\"terminal\":0},{\"name\":\"scrape\",\"last_seen\":\"2020-04-28T18:06:39.58849766+02:00\",\"status\":\"offline\",\"type\":\"test\",\"terminal\":0}]"
+  }
+]
+```
+
+
+## How to save config via agent
+Agent can be used to send configuration file for the [Export][export] service from cloud to gateway via MQTT.  
+Here is the example command:
+
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h localhost -p 1883  -m  "[{\"bn\":\"1:\", \"n\":\"config\", \"vs\":\"<config_file_path>, <file_cont_base64>\"}]"
```

Comment:
Replace `<file_cont_base64>` by `<file_content_base64>`. It hineders readility (`cont` is usually short for `continue`), 3 letters will not change things here.